src/modules/dashboard/staff/components/staff-form.vue

```diff
@@ -52,51 +55,77 @@ const props = defineProps({
   },
 });
 
+const formSchema = toTypedSchema(
+  z.object({
+    name: z.string().min(1, i18n.global.t('Required field')).max(50),
+    lastname: z
+      .string()
+      .min(1, i18n.global.t('Required field'))
+      .max(50),
+    email: z
+      .string()
+      .min(1, i18n.global.t('Required field'))
+      .email(i18n.global.t('Invalid email')),
+    phoneNumber: z
+      .string()
+      .min(1, i18n.global.t('Required field'))
+      .length(9, i18n.global.t('Phone number must be 9 digits'))
+      .regex(/^\d+$/, i18n.global.t('Only digits are allowed')),
+  }),
+);
+
 const staffStore = useStaffStore();
-const staffData = ref(new StaffData());
-const errors = ref<Record<string, string>>({});
+const { handleSubmit, errors, resetForm } = useForm({
+  validationSchema: formSchema,
+  initialValues: getFormInitialValues(),
+});
 
-onMounted(() => {
+function getFormInitialValues() {
   if (props.actionType === ActionType.EDIT) {
     const member = staffStore.getMember(props.staffId);
     if (member) {
-      staffData.value.setData(member);
+      return {
+        name: member.name,
+        lastname: member.lastname,
+        email: member.email,
+        phoneNumber: member.phone,
+        role: member.role || Roles.ASSISTANT,
+      };
```

Comment:
Jeżeli member ma tylko te same propsy jak przekazujesz tutaj to możesz zrobić 

`{ ...member, role: member.role || Roles.ASSISTANT }`

Jeśli nie, to przynajmniej zdestrukturyzowałbym membera aby zrobić shorthandy
`{ name, lastname, email, phone, role: role || Roles.ASSISTANT} ` - jedynie tam phoneNumber bedzie do poprawki